capstone/src/main/java/com/google/sps/servlets/ClubServlet.java

```diff
@@ -20,4 +36,59 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.setContentType("text/html;");
     response.getWriter().println(json);
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    String founderEmail = userService.getCurrentUser().getEmail();
+
+    // Check if club name is valid
+    Query query =
+        new Query("Club")
+            .setFilter(
+                new FilterPredicate(
+                    Constants.CLUB_NAME_PROP,
+                    FilterOperator.EQUAL,
+                    request.getParameter(Constants.CLUB_NAME_PROP)));
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery prepared = datastore.prepare(query);
+    response.setContentType("text/html;");
+    boolean isValid = Iterables.isEmpty(prepared.asIterable());
+
+    response.getWriter().println("<p>isValid = " + isValid + "</p>");
+    System.out.println("isValid = " + isValid);
+
+    String clubName = request.getParameter(Constants.CLUB_NAME_PROP);
+    String description = request.getParameter(Constants.DESCRIP_PROP);
+    String website = request.getParameter(Constants.WEBSITE_PROP);
+    BlobKey key = getBlobKey(request, Constants.LOGO_PROP);
+
+    Entity clubEntity = new Entity("Club", clubName);
+    clubEntity.setProperty(Constants.CLUB_NAME_PROP, clubName);
+    clubEntity.setProperty(Constants.DESCRIP_PROP, description);
+    clubEntity.setProperty(Constants.WEBSITE_PROP, website);
+    clubEntity.setProperty(Constants.MEMBER_PROP, ImmutableList.of(founderEmail));
+    clubEntity.setProperty(Constants.OFFICER_PROP, ImmutableList.of(founderEmail));
+    clubEntity.setProperty(Constants.ANNOUNCE_PROP, ImmutableList.of(""));
+    clubEntity.setProperty(Constants.LOGO_PROP, key);
+    response.sendRedirect("/club-registration.html");
```

Comment:
This works, it's also ok to allow the client to redirect once it receives the backend response.  This would be an easy way to handle error cases.

Reply:
Thank you! Redirecting seems to work well.